File: tools/conf/output/ts/CompetitionConfig.ts
class CompetitionConfig {
    private static _dic:Dictionary<CompetitionConfig> = new Dictionary<CompetitionConfig>();
    private static _arr:Array<CompetitionConfig> = new Array<CompetitionConfig>();

	 public Id :number ;
     public Type :number ;
     public ApartTime :number ;
     public OffsetTime :number ;
     public StopTime :number ;
     public FreshenTime :number ;
     public WaitingTime :number ;
     public Space :number ;
     public ProSpace :number ;
     public DataSize :number ;
     public DataPosition : Array<number>  =new Array<number>();
     public DataColor :string ;
     public DataDotColor :string ;
     public UprealRate :number ;
     public UprealValue :number ;
     public UpdashedRate :number ;
     public UpdashedValue :number ;
     public DownrealRate :number ;
     public DownrealValue :number ;
     public DowndashedRate :number ;
     public DowndashedValue :number ;
     public BetIds : Array<number>  =new Array<number>();
     public IntervalIds : Array<number>  =new Array<number>();
     public McId :number ;
     public Location : Array<number>  =new Array<number>();
     public Scale :number ;
     public ParticleId :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<CompetitionConfig>{
        return  CompetitionConfig._dic;
    }
    public static get Arr():Array<CompetitionConfig>{
        return CompetitionConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:CompetitionConfig = <CompetitionConfig>json[i];
            CompetitionConfig._arr.push(config);
            CompetitionConfig._dic.add(config.Id,config);
        }
    }
}

File: server/src/game/conf/CompetitionConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type CompetitionConfigElem struct {
	Id int
	Type int
	ApartTime float32
	OffsetTime float32
	StopTime float32
	FreshenTime float32
	WaitingTime float32
	Space int
	ProSpace int
	DataSize int
	DataPosition []int
	DataColor string
	DataDotColor string
	UprealRate float32
	UprealValue float32
	UpdashedRate float32
	UpdashedValue float32
	DownrealRate float32
	DownrealValue float32
	DowndashedRate float32
	DowndashedValue float32
	BetIds []int
	IntervalIds []int
	McId int
	Location []int
	Scale float32
	ParticleId int
	
}

type CompetitionConfig0 struct {
	Dic map[int]*CompetitionConfigElem
	Arr []*CompetitionConfigElem
}

var (
	CompetitionConfig CompetitionConfig0
)

func loadCompetitionConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	CompetitionConfig.Dic = make(map[int]*CompetitionConfigElem)
	CompetitionConfig.Arr = make([]*CompetitionConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &CompetitionConfig.Arr)
	for _,conf := range CompetitionConfig.Arr{
		CompetitionConfig.Dic[conf.Id] = conf
	}
	
}
